lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,707 @@
+    required: false
+  deletion_protection:
+    description:
+      - Indicates whether deletion protection for the ELB is enabled.
+    required: false
+    default: no
+    choices: [ 'yes', 'no' ]
+  idle_timeout:
+    description:
+      - The number of seconds to wait before an idle connection is closed.
+    required: false
+    default: 60
+  listeners:
+    description:
+      - A list of dicts containing listeners to attach to the ELB. See examples for a detail of the dict required.
+    required: false
+  name:
+    description:
+      - The name of the load balancer. This name must be unique within your AWS account, can have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, and must not begin or end with a hyphen.
+    required: true
+  purge_tags:
+    description:
+      - If yes, existing tags will be purged from the resource to match exactly what is defined by tags parameter. If the tag parameter is not set then tags will not be modified.
+    required: false
+    default: yes
+    choices: [ 'yes', 'no' ]
+  subnets:
+    description:
+      - A list of the IDs of the subnets to attach to the load balancer. You can specify only one subnet per Availability Zone. You must specify subnets from at least two Availability Zones. Required if state=present.
+    required: false
+  security_groups:
+    description:
+      - A list of the names or IDs of the security groups to assign to the load balancer. Required if state=present.
+    required: false
+    default: []
+  scheme:
+    description:
+      - Internet-facing or internal load balancer. An ELB scheme can not be modified after creation.
+    required: false
+    default: internet-facing
+    choices: [ 'internet-facing', 'internal' ]
+  state:
+    description:
+      - Create or destroy the load balancer.
+    required: true
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A dictionary of one or more tags to assign to the load balancer.
+    required: false
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create an ELB and attach a listener
+- name: elb_application_lb
+    name: myelb
+    security_groups:
+      - sg-12345678
+      - my-sec-group
+    subnets:
+      - subnet-012345678
+      - subnet-abcdef000
+    listeners:
+      - protocol: http # Required. The protocol for connections from clients to the load balancer (http or https).
+        port: 80 # Required. The port on which the load balancer is listening.
+        ssl_policy: # The security policy that defines which ciphers and protocols are supported. The default is the current predefined security policy.
+        certificates: # The ARN of the certificate (only one certficate ARN should be provided)
+        default_actions:
+          - type: forward # Required. Only 'forward' is accepted at this time
```

Comment:
Can't default this as it's not actually an Ansilble parameter.  It's a child of `listeners`.